aws-java-sdk-core/src/main/java/com/amazonaws/auth/InstanceProfileCredentialsProvider.java

```diff
@@ -121,6 +138,7 @@ protected void finalize() throws Throwable {
 
     @Override
     public AWSCredentials getCredentials() {
+        getCredentialsCalled = true;
         return credentialsFetcher.getCredentials();
     }
```

Comment:
Is it sufficient that `getCredentials()` was called? A stronger statement would be '`getCredentials()` was called and succeeded' such as
```java
     public AWSCredentials getCredentials() {
          AWSCredentials  creds = credentialsFetcher.getCredentials();
          getCredentialsCalled = true;
          return creds;
      }
```